WIP:Add k8s deployment yaml files for audit purpose only

benjaminhuo opened the pull request:
Signed-off-by: Benjamin <benjamin@yunify.com>

<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines in the [CONTRIBUTING.md](CONTRIBUTING.md) file and learn how to compile Falco from source [here](https://falco.org/docs/source).
2. Please label this pull request according to what type of issue you are addressing.
5. Please add a release note!
6. If the PR is unfinished while opening it specify a wip in the title before the actual title, for example, "wip: my awesome feature"
-->

**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

> /kind bug

/kind cleanup

> /kind design

/kind documentation

> /kind failing-test

/kind feature

> /kind flaky-test

> If contributing rules or changes to rules, please make sure to also uncomment one of the following line:

> /kind rule-update

> /kind rule-create

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

> /area engine

> /area rules

> /area deployment

/area integrations

> /area examples

**What this PR does / why we need it**:
Daemonset is more than enough for just monitoring k8s audit event, a deployment should be enough.

**Which issue(s) this PR fixes**:
https://github.com/falcosecurity/falco/issues/725

<!--
Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
If PR is `kind/failing-tests` or `kind/flaky-test`, please post the related issues/tests in a comment and do not use `Fixes`.
-->

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
Yes

<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, prepend the string "action required:".
For example, `action required: change the API interface of the rule engine`.
-->

```release-note
Add deployment yaml files instead of daemonset for k8s audit only purpose.
```

poiana commented:
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **mstemm**

If they are not already assigned, you can assign the PR to them by writing `/assign @mstemm` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=falcosecurity%2Ffalco).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/falcosecurity/falco/blob/dev/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":["mstemm"]} -->

benjaminhuo commented:
/assign @mstemm

benjaminhuo commented:
> Thanks for sending this PR, really appreciated! 🤗
> 
> A few initial observations.
> 
> Currently we have `integrations/k8s-using-daemonset` directory.
> The `integrations` directory probably needs a re-organization.
> Just to be in pair with the current structure can we put this integration under `integrations/k8s-using-deployment`?
> 
> Furthermore atm I think that `falco-k8s-audit.yaml` config file is not something that have to be in the root of the source.

I've made corresponding changes per your suggestions.
One more question, I tried to remove more options from falco.yaml for k8s audit only but I'm not sure which options I should remove and would like to know your suggestions :)

fntlnz commented:
@mfdii PTAL

leodido commented:
/hold

blocked by #730

fntlnz commented:
I think that this can now be accomplished since  #730 had been done (but not released yet).
@benjaminhuo let me know if you need assistance with it.

benjaminhuo commented:
@fntlnz I've modified my PR to add the new --disable-source flag and removed some extra config and rule files.

Would you help to review this ?

Thanks
Ben

fntlnz commented:
Looking again! :eyes: 

/assign @fntlnz 
/assign @leodido 
/un-assign @mstemm

fntlnz commented on integrations/k8s-using-deployment/README.md:
```diff
@@ -0,0 +1,62 @@
+# Example Kubernetes Deployments for Sysdig Falco
+
+This directory gives you the required YAML files to stand up Sysdig Falco on Kubernetes only for audit purpose as a Deployment.
```
```suggestion
This directory gives you the required YAML files to stand up Falco on Kubernetes only for audit purpose as a Deployment.
```

fntlnz commented on integrations/k8s-using-deployment/README.md:
```diff
@@ -0,0 +1,62 @@
+# Example Kubernetes Deployments for Sysdig Falco
```
```suggestion
# Example Kubernetes Deployments for Falco
```

fntlnz commented on integrations/k8s-using-deployment/falco.yaml:
```diff
@@ -0,0 +1,167 @@
+#
```
This file must be ignored, in the instructions you are telling to copy it.

fntlnz replied:
No this must be kept for now, it differs from the default one and we have no way of disabling the default rules right now so a specific one seems good.

fntlnz commented:
ok @benjaminhuo we are just waiting for you to accept the renaming changes then we can merge this!

poiana commented:
Thanks for your pull request. Before we can look at it, you'll need to add a 'DCO signoff' to your commits.

:memo: **Please follow instructions in the [contributing guide](https://github.com/falcosecurity/falco/blob/master/CONTRIBUTING.md) to update your commits with the DCO**

Full details of the Developer Certificate of Origin can be found at [developercertificate.org](https://developercertificate.org/).

**The list of commits missing DCO signoff**:

- [d26d7cb](https://github.com/falcosecurity/falco/commits/d26d7cb02fae56fb64a29f026fb25b34b904d96d) Update integrations/k8s-using-deployment/README.md

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

poiana commented:
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **fntlnz**

If they are not already assigned, you can assign the PR to them by writing `/assign @fntlnz` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=falcosecurity%2Ffalco).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/falcosecurity/falco/blob/dev/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":["fntlnz"]} -->

poiana commented:
Thanks for your pull request. Before we can look at it, you'll need to add a 'DCO signoff' to your commits.

:memo: **Please follow instructions in the [contributing guide](https://github.com/falcosecurity/falco/blob/master/CONTRIBUTING.md) to update your commits with the DCO**

Full details of the Developer Certificate of Origin can be found at [developercertificate.org](https://developercertificate.org/).

**The list of commits missing DCO signoff**:

- [d26d7cb](https://github.com/falcosecurity/falco/commits/d26d7cb02fae56fb64a29f026fb25b34b904d96d) Update integrations/k8s-using-deployment/README.md
- [523bfd1](https://github.com/falcosecurity/falco/commits/523bfd15ce1ef2ab0ca75082ee3feba72e72d675) Update integrations/k8s-using-deployment/README.md

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

benjaminhuo commented:
@leodido @fntlnz , Sorry I missed your comments and I've accept the change now.

Regards,
Ben

benjaminhuo commented:
I've merged your suggestions and force pushed the new changes

poiana commented:
LGTM label has been added.  <details>Git tree hash: 6ab2904d1126e329e76904da8d5d9f49a57821ff</details>

poiana commented:
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/falcosecurity/falco/pull/729#pullrequestreview-298921796" title="Approved">leodido</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=falcosecurity%2Ffalco).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/falcosecurity/falco/blob/dev/OWNERS)~~ [leodido]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":[]} -->

leodido commented:
/hold cancel
